### PR TITLE
update Merge content

### LINKED ISSUE #1001: Make clarifications in conceptual merge documentation
Clarify conceptual merge docs according to feedback in https://github.com/ConsenSys/doc.teku/pull/352

### PR DESCRIPTION
**Pull request by alexandratran**
Signed-off-by: Alexandra Tran <alexandra.tran@consensys.net>

## Pull request checklist

Use the following list to make sure your PR fits the Besu doc quality standard.

### Before creating the pull request

Make sure that:

- [x] [all commits in this PR are signed off for the DCO](https://wiki.hyperledger.org/display/BESU/DCO).
- [x] you read the [contribution guidelines](https://wiki.hyperledger.org/display/BESU/Contributing+to+documentation).
- [x] you have [tested your changes locally](https://wiki.hyperledger.org/display/BESU/MkDocs+And+Markdown+Guide#MkDocsAndMarkdownGuide-PreviewTheDocumentation) before submitting them to the community for review.

### After creating your pull request and tests finished

Make sure that:

- [ ] you fixed all the issues raised by the tests, if any.
- [ ] you verified the rendering of your changes on [ReadTheDocs.org PR preview](https://wiki.hyperledger.org/display/BESU/MkDocs+And+Markdown+Guide#MkDocsAndMarkdownGuide-PreviewwithReadTheDocs)
  and updated the testing link (see [Testing](#testing)).

## Describe the change

Update merge content to integrate feedback given in the [Teku docs PR](https://github.com/ConsenSys/doc.teku/pull/352).

## Issue fixed

fixes #1001 

### For content changes

- [x] Doc content
- [ ] Doc pages organisation

### For tools changes

- [ ] CircleCI workflow
- [ ] Build and QA tools (lint, vale,…)
- [ ] MkDocs templates
- [ ] MkDocs configuration
- [ ] Python dependencies
- [ ] Node dependencies and JavaScript
- [ ] ReadTheDocs configuration
- [ ] GitHub integration

### CODE REVIEW

**Review by rolandtyler — APPROVED** [review]
lgtm

File: docs/Concepts/Merge.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ description: What is the Merge?
 The Ethereum upgrade known as [The Merge](https://ethereum.org/en/upgrades/merge/) will merge the [Beacon Chain] into
 Ethereum Mainnet, turning Mainnet into a combination of an
 [execution layer and consensus layer](#execution-and-consensus-clients).
-The Merge will transition Mainnet from proof of work to [proof of stake consensus](Consensus-Protocols/Proof-of-Stake.md).
+The Merge will transition Mainnet from proof of work to [proof of stake consensus](POS-LINK).
```
Comment (bgravenorst) [inline docs/Concepts/Merge.md:10]: The Merge transitions

**Review by bgravenorst — APPROVED** [review]
LGTM

**Comment by NicolasMassart** [comment]
If you still have to update the `POS_LINK` with the actual value, you can make this PR Draft.

**Pull request merged**